crates/symbolicator-service/src/services/symbolication/mod.rs

```diff
@@ -240,15 +256,27 @@ fn symbolicate_frame(
     frame.package = lookup_result.object_info.raw.code_file.clone();
 
     match lookup_result.cache {
-        Ok(CacheFileEntry::SymCache(symcache)) => symbolicate_native_frame(
-            symcache.get(),
-            lookup_result,
-            registers,
-            signal,
-            frame,
-            index,
-            adjustment,
-        ),
+        Ok(CacheFileEntry::SymCache(symcache)) => {
+            let symcache = symcache.get();
+            let relative_addr = get_relative_caller_addr(
+                symcache,
+                &lookup_result,
+                registers,
+                signal,
+                index,
+                adjustment,
+            );
+            relative_addr.and_then(|addr| {
+                symbolicate_native_frame(
+                    demangle_cache,
+                    symcache,
+                    lookup_result,
+                    addr,
+                    frame,
+                    index,
+                )
+            })
```

Comment:
Nit: IMO this is nicer with `?`.

```suggestion
            )?;

            symbolicate_native_frame(
                demangle_cache,
                symcache,
                lookup_result,
                relative_addr,
                frame,
                index,
            )
```